lib/pardot/authentication.rb

```diff
@@ -0,0 +1,27 @@
+module Pardot
+  module Authentication
+
+    def authenticate
+      resp = post 'login', nil, nil, nil, email: @email, password: @password, user_key: @user_key
+      update_version(resp['version']) if resp && resp['version']
+      @api_key = resp && resp['api_key']
+    end
+
+    def authenticated?
+      @api_key != nil
```

Comment:
May be paranoid, but `@api_key.present?` would also return `false` for `false`, `''`, as well as `nil`

Reply:
Fixed (or deflected) by using the gem instead of the putting the code in lib. 😄